src/test/java/com/exasol/adapter/dialects/exasol/ExasolSqlDialectTest.java

```diff
@@ -93,55 +106,65 @@ private SqlNode getTestSqlNode() {
         final List<ColumnMetadata> columns = new ArrayList<>();
+        final ColumnAdapterNotesJsonConverter converter = ColumnAdapterNotesJsonConverter.getInstance();
         columns.add(ColumnMetadata.builder().name("USER_ID")
-              .adapterNotes(ColumnAdapterNotes.serialize(new ColumnAdapterNotes(3, "DECIMAL")))
-              .type(DataType.createDecimal(18, 0)).nullable(true).identity(false).defaultValue("").comment("").build());
-        columns.add(ColumnMetadata.builder().name("URL")
-              .adapterNotes(ColumnAdapterNotes.serialize(new ColumnAdapterNotes(12, "VARCHAR")))
-              .type(DataType.createVarChar(10000, DataType.ExaCharset.UTF8)).nullable(true).identity(false)
-              .defaultValue("").comment("").build());
+                .adapterNotes(converter.convertToJson(new ColumnAdapterNotes(3))).type(DataType.createDecimal(18, 0))
+                .nullable(true).identity(false).defaultValue("").comment("").build());
+        columns.add(
+                ColumnMetadata.builder().name("URL").adapterNotes(converter.convertToJson(new ColumnAdapterNotes(12)))
+                        .type(DataType.createVarChar(10000, DataType.ExaCharset.UTF8)).nullable(true).identity(false)
+                        .defaultValue("").comment("").build());
         return new TableMetadata("CLICKS", "", columns, "");
     }
 
-    @CsvSource({"FALSE, FALSE, JDBC", //
-          "TRUE, FALSE, LOCAL", //
-          "FALSE, TRUE, EXA"})
-    @ParameterizedTest
-    void testGetImportTypeLocal(final String local, final String fromExasol, final String expectedImportType) {
-        this.rawProperties.put(IS_LOCAL_PROPERTY, local);
-        this.rawProperties.put(EXASOL_IMPORT_PROPERTY, fromExasol);
-        final ExasolSqlDialect dialect = new ExasolSqlDialect(null, new AdapterProperties(this.rawProperties));
-        assertThat(dialect.getImportType().toString(), equalTo(expectedImportType));
-    }
-
     @Test
-    void testMetadataReaderClass() {
+    void testCreateRemoteMetadataReader(@Mock final Connection connectionMock) throws SQLException {
+        when(this.connectionFactoryMock.getConnection()).thenReturn(connectionMock);
         assertThat(getMethodReturnViaReflection(this.dialect, "createRemoteMetadataReader"),
-              instanceOf(ExasolMetadataReader.class));
+                instanceOf(ExasolMetadataReader.class));
     }
 
     @Test
-    void testQueryRewriterClass() {
+    void testCreateJdbcQueryRewriter(@Mock final Connection connectionMock) throws SQLException {
+        when(this.connectionFactoryMock.getConnection()).thenReturn(connectionMock);
         assertThat(getMethodReturnViaReflection(this.dialect, "createQueryRewriter"),
-              instanceOf(ExasolQueryRewriter.class));
+                instanceOf(ExasolJdbcQueryRewriter.class));
+    }
+
+    @Test
+    void testCreateLocalQueryRewriter(@Mock final Connection connectionMock) throws SQLException {
```

Comment:
Unused parameter and exception is not thrown

Reply:
Fixed.